### PR TITLE
Enable app directory as alternative to file

### PR DESCRIPTION
**Pull request by niedzielski**
Make the project template more encouraging of multi-file projects by
moving app.js to app/index.js and updating the require.

### CODE REVIEW

**Comment by fletchto99** [comment]
I'm guessing doing something like this would probably require CloudPebble to also be updated to adjust for the new layout. Just a heads up @Meiguro (if you accept this)

**Comment by niedzielski** [comment]
I believe this change is backwards backwards compatible.

**Comment by Meiguro** [comment]
@fletchto99 is referring to the backwards compatibility of the project directory of Pebble.js projects on CloudPebble whose filesystem is not fully exposed.

@Katharine or @Spacerat, if you have time, how would this affect existing CloudPebble projects? I can take a look too if you have your hands tied.

**Comment by Meiguro** [comment]
I'm silly, it is backwards compatible because it is using loader.js' multiple extension fallback, but having others double check would be good.

**Comment by Meiguro** [comment]
The change is good, but it looks like @fletchto99 was right, CloudPebble will need to be updated to handle the new location of the root file, probably by trying both app.js and app/index.js. When I took a second look, I didn't notice the app.js file was also moved, so I had thought it was fully compatible.

> at cloudpebble/ide/api/project.py#L179
> [f.save_file(open('{}/src/js/app.js'.format(settings.PEBBLEJS_ROOT)).read())](https://github.com/pebble/cloudpebble/blob/1863a319380052b9ac7f5740b6702b9664c09fc7/ide/api/project.py#L179)

I can look into making a PR against CloudPebble for this.

**Comment by niedzielski** [comment]
@Meiguro, rats, I thought the way the loader worked was to consider any of `['?', '?.js', '?.json', '?/index.js']` extensions. So, `require('./app')` would match `app.js` if it exists, otherwise `app/index.js`. It seems to work on my system both ways (as is and reverting the file move).

**Comment by Katharine** [comment]
Note that CloudPebble doesn't generally much care for directories, and I'm disinclined to have any default template use them.

**Comment by Meiguro** [comment]
@niedzielski No, you got that part correct. The change to require just `'app'` is good, that is actually more CommonJS compliant. The are just two issues, creating a project on CloudPebble that I caught, and the idea of directories with regards to CloudPebble that Katharine brings up. If you personally need this change and don't want to change the way CloudPebble would handle pebble.js projects, you can revert all changes except the change to require, and it would be good for merging.

app.js is in `src/js` because I didn't want to deeply nest the user's javascript further. `src/js` is already one level deeper than a regular non-pebble javascript project. I tried to make the deeper folders make sense and be very little intrusive, but if they are causing problems, I would like to know. I can always move them deeper and add another require search path. Outside of `main.js` and `loader.js`, all other file names in `src/js` are probably not going to be used in the future, so users are free to place their multi-file application logic there.

**Comment by niedzielski** [comment]
@Meiguro, thanks for the insights. That sounds good to me! I made a small change to the readme to hint at alternative app structures. I can revert that if you don't like it. Let me know if you want me to squash these commits or change anything.

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Pebble.js applications run on your phone. They have access to all the resources
 
    This option allows you to customize Pebble.js. Follow the [Pebble SDK installation instructions](https://developer.pebble.com/sdk/install/) to install the SDK on your computer and [fork this project](http://github.com/pebble/pebblejs) on Github. 
    
-   The main entry point for your application is in the `src/js/app.js` file.
+   The main entry point for your application is in the `src/js/app.js` file. For projects with multiple files, you may use `src/app/index.js` instead and create new files under `src/app`.
```
Comment (Meiguro) [inline README.md]: Did you mean `src/js/app/index.js` and `src/js/app`? Otherwise looks good.

Comment (niedzielski) [inline README.md]: Ok, think I got it now :)

**Comment by Meiguro** [comment]
Thanks for this!

**Pull request merged**